crates/sui-indexer/src/apis/governance_api.rs

```diff
@@ -205,104 +165,100 @@ impl<T: R2D2Connection + 'static> GovernanceReadApi<T> {
-                sui_types::error::SuiError::ObjectDeserializationError {
-                    error: e.to_string(),
-                }
-            })?;
-            let inactive_pools_id = system_state_summary.inactive_pools_id;
-            let validator = self
-                .inner
-                .spawn_blocking(move |this| {
-                    sui_types::sui_system_state::get_validator_from_table(
-                        &this,
-                        inactive_pools_id,
-                        &pool_id,
-                    )
-                })
-                .await?;
-            tables.push((
-                validator.sui_address,
-                validator.staking_pool_id,
-                validator.exchange_rates_id,
-                validator.exchange_rates_size,
-                false,
-            ));
-        }
-
-        let mut exchange_rates = vec![];
-        // Get exchange rates for each validator
-        for (address, pool_id, exchange_rates_id, exchange_rates_size, active) in tables {
-            let mut rates = vec![];
-            for df in self
-                .inner
-                .get_dynamic_fields_raw_in_blocking_task(
-                    exchange_rates_id,
-                    None,
-                    exchange_rates_size as usize,
-                )
-                .await?
-            {
-                let dynamic_field = df
-                    .to_dynamic_field::<EpochId, PoolTokenExchangeRate>()
-                    .ok_or_else(|| sui_types::error::SuiError::ObjectDeserializationError {
-                        error: "dynamic field malformed".to_owned(),
-                    })?;
+            let dynamic_field = df
+                .to_dynamic_field::<EpochId, PoolTokenExchangeRate>()
+                .ok_or_else(|| sui_types::error::SuiError::ObjectDeserializationError {
+                    error: "dynamic field malformed".to_owned(),
+                })?;
 
-                rates.push((dynamic_field.name, dynamic_field.value));
-            }
+            rates.push((dynamic_field.name, dynamic_field.value));
+        }
 
-            rates.sort_by(|(a, _), (b, _)| a.cmp(b).reverse());
+        rates.sort_by(|(a, _), (b, _)| a.cmp(b).reverse());
 
-            exchange_rates.push(ValidatorExchangeRates {
-                address,
-                pool_id,
-                active,
-                rates,
-            });
-        }
-        Ok(exchange_rates)
+        exchange_rates.push(ValidatorExchangeRates {
+            address,
+            pool_id,
+            active,
+            rates,
+        });
     }
-}
-
-/// Cache a map representing the validators' APYs for this epoch
-#[cached(
-    type = "SizedCache<EpochId, BTreeMap<SuiAddress, f64>>",
-    create = "{ SizedCache::with_size(1) }",
-    convert = " {apys.epoch} "
-)]
-fn validators_apys_map(apys: ValidatorApys) -> BTreeMap<SuiAddress, f64> {
```

Comment:
and similarly, do we need to get rid of the map?

Reply:
The map is pretty much useless, because it will always return the latest APY for that validator.  I will rework this in a separate PR.